### PR TITLE
CustomPreprocessBuilder

### PR DESCRIPTION
**Pull request by GDGRhinox**
### Added
- Abstract base CustomProcessBuildWithReport as a replacement for IPreprocessBuildWithReport/IPostprocessBuildWithReport which can actually handle build failures

### Fixes
- CustomProjectSettingsBuilder now correctly cleans up state on build failure

### CODE REVIEW

File: Assets/Utilities/Editor/Helpers/ICustomPreProcessBuilder.cs [hunk 1]
```diff
@@ -0,0 +1,45 @@
+﻿using UnityEditor;
+using UnityEditor.Build;
+using UnityEditor.Build.Reporting;
+
+namespace Assets.Utilities.Editor.Helpers
```
Comment (GDGRhinox) [inline Assets/Utilities/Editor/Helpers/ICustomPreProcessBuilder.cs:5]: Namespace is inconsistent

File: Assets/Utilities/Editor/Helpers/ICustomPreProcessBuilder.cs [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+{
+    public abstract class CustomPreProcessBuilderWithErrorReport : IPreprocessBuildWithReport, IPostprocessBuildWithReport
```
Comment (GDGRhinox) [inline Assets/Utilities/Editor/Helpers/ICustomPreProcessBuilder.cs:7]: - Rename class to CustomProcessBuildWithReport
- Rename file to be consistent with the class

File: Assets/Utilities/Editor/Helpers/ICustomPreProcessBuilder.cs [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    {
+        private static bool _BuildDone;
```
Comment (GDGRhinox) [inline Assets/Utilities/Editor/Helpers/ICustomPreProcessBuilder.cs:9]: Code style + rename -> _buildFinished

File: Packages/manifest.json [hunk 4]
```diff
@@ -18,7 +18,7 @@
     "com.unity.editorcoroutines": "1.0.0",
     "com.unity.ext.nunit": "1.0.6",
```
Comment (GDGRhinox) [inline Packages/manifest.json:19]: Missing package.json update

**Comment by GlennQuintyn** [comment]
Remarks have been applied.

**Pull request merged**